airflow/contrib/operators/docker_swarm_operator.py

```diff
@@ -123,11 +129,43 @@ def _run_image(self):
 
         self.log.info('Service started: %s', str(self.service))
 
-        status = None
         # wait for the service to start the task
         while not self.cli.tasks(filters={'service': self.service['ID']}):
             continue
-        while True:
+
+        logs = self.cli.service_logs(
+            self.service['ID'], follow=True, stdout=True, stderr=True, is_tty=self.tty
+        )
+        line = ''
+        _stream_logs = self.enable_logging  # Status of the service_logs' generator
+        while True:  # pylint: disable=too-many-nested-blocks
+            if self.enable_logging:
```

Comment:
Could you please break this up into smaller functions? Easier for possible testing later.

Reply:
I had tried that earlier but this whole block makes sense only if put together. The `while True` has 2 responsibilities - log things and check if the service still running - but I don't see any straightforward way to decouple them.
Also, this block waits for the service to stop which doesn't make sens without the part where the service is started.

Do you have any particular ideas in mind?

Reply:
Added a comment to the code, if that helps with the readability.

Reply:
@dimberman Does this work for you? Or do you have any other suggestions?

Reply:
@dimberman I have addressed this now - please have a look.